message-store/message-store.bal

```diff
@@ -0,0 +1,21 @@
+import ballerina/http;
+
+type GeoCodeResponse record {|
+    json results;
+|};
+
+final http:Client geoCodingClient = check new ("http://api.maps.googleapis.com.balmock.io");
+final http:Client firebaseClient = check new ("http://api.mapsproject.firebase.com.balmock.io");
+
+service /api on new http:Listener(8080) {
+    
+    resource function get location(string address) returns GeoCodeResponse|error {
+        GeoCodeResponse|error storedGeocode = firebaseClient->/location/[address]/["location.json"]();
+        if storedGeocode !is error {
+            return storedGeocode;
+        }
+        GeoCodeResponse geocode = check geoCodingClient->/maps/api/geocode/'json(place=address);
+        json _ = check firebaseClient->/location/["location.json"].post({address, ...geocode});
```

Comment:
Why address send in body? should it be in URL, when you get it (line 13), it's in the URL.

Reply:
No this the way to post data.
 https://firebase.google.com/docs/reference/rest/database#section-post

Reply:
Looks like `POST` is the wrong thing to do. https://firebase.google.com/docs/database/rest/save-data#section-post
it **push**es with a random id. I think we need `PUT`.

Reply:
Updated.